client/wp-admin-scripts/onboarding-tax-notice/index.js

```diff
@@ -28,30 +27,36 @@ const saveCompleted = () => {
 const showTaxCompletionNotice = () => {
 	const saveButton = document.querySelector( '.woocommerce-save-button' );
 
-	if ( saveButton.classList.contains( 'is-clicked' ) ) {
+	if ( saveButton.classList.contains( 'has-tax' ) ) {
 		return;
 	}
 
-	saveButton.classList.add( 'is-clicked' );
-	saveCompleted().then( () =>
-		dispatch( 'core/notices' ).createSuccessNotice(
-			__( "You've added your first tax rate!", 'woocommerce-admin' ),
-			{
-				id: 'WOOCOMMERCE_ONBOARDING_TAX_NOTICE',
-				actions: [
-					{
-						url: getAdminLink( 'admin.php?page=wc-admin' ),
-						label: __( 'Continue setup.', 'woocommerce-admin' ),
-					},
-				],
+	saveCompleted().then( () => {
+		// Check if a row was added successfully after WooCommerce removes invalid rows.
+		setTimeout( () => {
```

Comment:
I think we could prevent the use of `setTimeout` here by checking if the `saveButton` is disabled again.
When trying this locally it seemed that the saveCompleted function didn't work accurately anymore, it never hit the case where it found `.blockUI.blockOverlay` .
We could switch that to `if (saveButton.disabled) {`.
It looked like the invalid rows where removed after the save button was set back to disabled as well.